tsdb/head_wal.go

```diff
@@ -370,14 +327,38 @@ Outer:
 	if unknownRefs.Load() > 0 || unknownExemplarRefs.Load() > 0 || unknownMetadataRefs.Load() > 0 {
 		level.Warn(h.logger).Log("msg", "Unknown series references", "samples", unknownRefs.Load(), "exemplars", unknownExemplarRefs.Load(), "metadata", unknownMetadataRefs.Load())
 	}
-	if mmapOverlappingChunks > 0 {
-		level.Info(h.logger).Log("msg", "Overlapping m-map chunks on duplicate series records", "count", mmapOverlappingChunks)
+	if count := mmapOverlappingChunks.Load(); count > 0 {
+		level.Info(h.logger).Log("msg", "Overlapping m-map chunks on duplicate series records", "count", count)
 	}
 	return nil
 }
 
 // resetSeriesWithMMappedChunks is only used during the WAL replay.
-func (h *Head) resetSeriesWithMMappedChunks(mSeries *memSeries, mmc []*mmappedChunk) {
+func (h *Head) resetSeriesWithMMappedChunks(mSeries *memSeries, mmc []*mmappedChunk, maybeOverlappedSeriesRef chunks.HeadSeriesRef) (overlapped bool) {
+	if maybeOverlappedSeriesRef != 0 && mSeries.ref != maybeOverlappedSeriesRef {
```

Comment:
We dont need `maybeOverlappedSeriesRef != 0` check because it will never be false in our use.